npm/util/util.go

```diff
@@ -98,11 +102,15 @@ func GetHashedName(name string) string {
 // returns -1, 0, 1 if firstVer smaller, equals, bigger than secondVer respectively.
 // returns -2 for error.
 func CompareK8sVer(firstVer *version.Info, secondVer *version.Info) int {
-	v1, err := semver.NewVersion(firstVer.Major+firstVer.Minor)
+	minorVers := re.FindAllString(firstVer.Minor, -1)
+	if len(minorVers) < 1 {
+		return -2
+	}
+	v1, err := semver.NewVersion(firstVer.Major+"."+minorVers[0])
 	if err != nil {
 		return -2
 	}
-	v2, err := semver.NewVersion(secondVer.Major+secondVer.Minor)
+	v2, err := semver.NewVersion(secondVer.Major+"."+secondVer.Minor)
```

Comment:
as a utility function, I think you should not assume the caller always give a right format in the secondVer.